Sources/NIOWebSocket/Base64.swift

```diff
@@ -33,80 +33,66 @@ internal struct Base64 {
     var input = bytes.makeIterator()
+    var offset = 0
   
     while let firstByte = input.next() {
       let secondByte = input.next()
       let thirdByte = input.next()
       
-      let firstChar  = Base64.encode(alphabet: alphabet, firstByte: firstByte)
-      let secondChar = Base64.encode(alphabet: alphabet, firstByte: firstByte, secondByte: secondByte)
-      let thirdChar  = Base64.encode(alphabet: alphabet, secondByte: secondByte, thirdByte: thirdByte)
-      let forthChar  = Base64.encode(alphabet: alphabet, thirdByte: thirdByte)
-      
-      outputBytes.append(firstChar)
-      outputBytes.append(secondChar)
-      outputBytes.append(thirdChar)
-      outputBytes.append(forthChar)
+      outputBuffer[offset] = Base64.encode(alphabet: alphabet, firstByte: firstByte)
+      outputBuffer[offset + 1] = Base64.encode(alphabet: alphabet, firstByte: firstByte, secondByte: secondByte)
+      outputBuffer[offset + 2] = Base64.encode(alphabet: alphabet, secondByte: secondByte, thirdByte: thirdByte)
+      outputBuffer[offset + 3] = Base64.encode(alphabet: alphabet, thirdByte: thirdByte)
+      offset += 4
     }
+    
+    // last char has to be a `0` in a zero terminated C string
+    outputBuffer[offset] = 0
 
-    return String(decoding: outputBytes, as: Unicode.UTF8.self)
+    // it is safe to force unwrap the baseAddress since we know that the output buffer is not empty
+    return String(cString: outputBuffer.baseAddress!)
   }
   
   // MARK: Internal
   
   // The base64 unicode table.
   @usableFromInline
-  static let encodeBase64: [UInt8] = [
-    UInt8(ascii: "A"), UInt8(ascii: "B"), UInt8(ascii: "C"), UInt8(ascii: "D"),
-    UInt8(ascii: "E"), UInt8(ascii: "F"), UInt8(ascii: "G"), UInt8(ascii: "H"),
-    UInt8(ascii: "I"), UInt8(ascii: "J"), UInt8(ascii: "K"), UInt8(ascii: "L"),
-    UInt8(ascii: "M"), UInt8(ascii: "N"), UInt8(ascii: "O"), UInt8(ascii: "P"),
-    UInt8(ascii: "Q"), UInt8(ascii: "R"), UInt8(ascii: "S"), UInt8(ascii: "T"),
-    UInt8(ascii: "U"), UInt8(ascii: "V"), UInt8(ascii: "W"), UInt8(ascii: "X"),
-    UInt8(ascii: "Y"), UInt8(ascii: "Z"), UInt8(ascii: "a"), UInt8(ascii: "b"),
-    UInt8(ascii: "c"), UInt8(ascii: "d"), UInt8(ascii: "e"), UInt8(ascii: "f"),
-    UInt8(ascii: "g"), UInt8(ascii: "h"), UInt8(ascii: "i"), UInt8(ascii: "j"),
-    UInt8(ascii: "k"), UInt8(ascii: "l"), UInt8(ascii: "m"), UInt8(ascii: "n"),
-    UInt8(ascii: "o"), UInt8(ascii: "p"), UInt8(ascii: "q"), UInt8(ascii: "r"),
-    UInt8(ascii: "s"), UInt8(ascii: "t"), UInt8(ascii: "u"), UInt8(ascii: "v"),
-    UInt8(ascii: "w"), UInt8(ascii: "x"), UInt8(ascii: "y"), UInt8(ascii: "z"),
-    UInt8(ascii: "0"), UInt8(ascii: "1"), UInt8(ascii: "2"), UInt8(ascii: "3"),
-    UInt8(ascii: "4"), UInt8(ascii: "5"), UInt8(ascii: "6"), UInt8(ascii: "7"),
-    UInt8(ascii: "8"), UInt8(ascii: "9"), UInt8(ascii: "+"), UInt8(ascii: "/"),
-  ]
+  static let encodeBase64: StaticString = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/"
```

Comment:
I’m _highly_ suspicious of this change being helpful. There are very few data types in the standard library cheaper than `Array`. There _is_ one advantage, which is that `String` can be embedded directly in the binary where `Array<UInt8>` tends to fail to be embedded: https://bugs.swift.org/browse/SR-14102. 

Regardless, I’d like to pull this change out to a separate one so we can get an idea of exactly how meaningful it is.

Reply:
This being a `StaticString` it is definitely embedded, but I agree that it is better to put it in a separate PR.
I will remove this change from this PR.